src/cephadm/cephadm

```diff
@@ -6060,7 +6077,12 @@ def command_adopt(ctx):
     # type: (CephadmContext) -> None
 
     if not ctx.skip_pull:
-        _pull_image(ctx, ctx.image)
+        try:
+            _pull_image(ctx, ctx.image)
+        except UnauthorizedRegistryError:
+            err_str = 'Failed to pull container image. Check that correct registry credentials are provided by `cephadm registry-login <url> <username> <password>` or supply login info via -i <file>'
```

Comment:
```suggestion
            err_str = 'Failed to pull container image. Host may not be logged into container registry. Try  `cephadm registry-login --regsitry-url <url> --registry-username <username> --registry-password <password>` or supply login info via a json file with `cephadm resgistry-login --registry-json <file>`'
```

You actually need to explicitly provide the flags for `cephadm registry-login`. Positional args don't work in the binary.